mxnet-bot/LabelBotAddLabels/label_bot.py

```diff
@@ -0,0 +1,143 @@
+    assert obj in set(["issues", "labels"]), "Invalid Input!"
+    url = 'https://api.github.com/repos/{}/{}'.format(REPO, obj)
+    if obj == 'issues':
+        response = requests.get(url, {'state': state},
+                                auth=AUTH)
+    else:
+        response = requests.get(url, auth=AUTH)
+    assert response.status_code == 200, response.status_code
+    if "link" not in response.headers:
+        return 1
+    # response.headers['link'] will looks like:
+    # <https://api.github.com/repositories/34864402/issues?state=all&page=387>; rel="last"
+    # In this case we need to extrac '387' as the count of pages
+    return int(clean_string(response.headers['link'], " ").split()[-3])
+
+
+# This method is to find comments which @mxnet-label-bot
+def find_notifications():
+    data = []
+    pages = count_pages("issues")
+    for page in range(1, pages+1):
+        url = 'https://api.github.com/repos/' + REPO + '/issues?page=' + str(page) \
+            + '&per_page=30'.format(repo=REPO)
+        response = requests.get(url,
+                                {'state': 'open',
+                                 'base': 'master',
+                                 'sort': 'created',
+                                 'direction': 'desc'},
+                                auth=AUTH)
+        for item in response.json():
+            # limit the amount of unlabeled issues per execution
+            if len(data) >= 50:
+                break
+            if "pull_request" in item:
+                continue
+            if not item['labels']:
+                if item['comments'] != 0:
+                    labels = []
+                    comments_url = "https://api.github.com/repos/{}/issues/{}/comments".format(REPO,item['number'])
+                    comments = requests.get(comments_url, auth=AUTH).json()
+                    for comment in comments:
+                        if "@mxnet-label-bot" in comment['body']:
+                            labels += tokenize(comment['body'])
+                            logger.info("issue: {}, comment: {}".format(str(item['number']),comment['body']))
+                    if labels != []:
+                        data.append({"issue": item['number'], "labels": labels})
+    return data
+
+
+# find all existing labels in a repo
+def all_labels():
+    pages = count_pages("labels")
+    all_labels = []
+    for page in range(1, pages+1):
+        url = 'https://api.github.com/repos/' + REPO + '/labels?page=' + str(page) \
+            + '&per_page=30'.format(repo=REPO)
+        response = requests.get(url, auth=AUTH)
+        for item in response.json():
+            all_labels.append(item['name'].lower())
+    logger.info("{} labels in total".format(str(len(all_labels))))
+    return set(all_labels)
+
+
+all_labels = all_labels()
+
+
+# add labels to one issue
+def add_github_labels(number, labels):
+    # number: the issue number
+    # labels: list of strings
```

Comment:
can we describe the semantics of labels? It's nice to have the type but what are they? the labels you want to set I assume. Also explain that uses the intersection with all labels in the function docstring.
Would be good to use python propper docstrings instead of # comments

Reply:
Docstrings are added